benefits/core/models.py

```diff
@@ -235,6 +381,7 @@ class EnrollmentFlow(models.Model):
         default=[EnrollmentMethods.DIGITAL, EnrollmentMethods.IN_PERSON],
         help_text="If the flow is supported by digital enrollment, in-person enrollment, or both",
     )
+    transit_agency = models.ForeignKey(TransitAgency, on_delete=models.PROTECT)
```

Comment:
The migration is failing for me; we need to update this field to allow nulls.

### Steps I'm using to test migration:
- Checkout `main` and reset database using sample fixtures -- this makes our database use the same schema as that on our Azure instances (dev, test, and prod)
   - `git checkout main`
   - `./bin/reset_db.sh`
- Checkout `refactor/transit-agency-enrollment-flow`
- Run `./bin/init.sh` to apply migration

Migration fails with:
```
django.db.utils.IntegrityError: NOT NULL constraint failed: new__core_enrollmentflow.transit_agency_id
```

`EnrollmentFlow.transit_agency` needs to allow nulls, i.e. pass in [`null=True`](https://docs.djangoproject.com/en/5.1/ref/models/fields/#null) when declaring it as a `ForeignKey` field. We should also pass in `blank=True` so that from the admin interface, the model form allows us to set a blank value and pass form validation.